### PR TITLE
Split VS Code debugger lesson into prereading and lesson

### PR DESCRIPTION
**Pull request by jmaddox19**

### CODE REVIEW

File: 00-programming-fundamentals/setting-up-vs-code-debugger.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      "cwd": "${workspaceRoot}",
+      "program": "${file}"
+    }
+  ]
+}
+```
+
+when you are finished, it should look approximately like this:
+
+![The functional part of the json is in a field called 'launch' within the settings json.](images/debug_settings_3.png "The functional part of the json is in a field called 'launch' within the settings json")
```
Comment (CheezItMan) [inline 00-programming-fundamentals/setting-up-vs-code-debugger.md]: I might highlight the comma on the line before `"launch": {`, just because someone unfamiliar with JSON might mistake it.

File: 00-programming-fundamentals/setting-up-vs-code-debugger.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+For every two developers, there are three opinions about the best environment for making code. So how do we pick? It all depends on how much help (some would say interference) a programmer wants moment to moment.
+
+On the most basic level, a Text Editor will do just that: let you edit the words and symbols on a page unfettered by suggestion text or drop downs. An Integrated Development Environment (IDE for short) on the other hand has opinions about how you should be doing your work: they may try and give you the name of the function you were about to type, help manage a source control tool, or (as we will do) help you debug your code.
+
+__Note: If anyone ever tells you that "real developers only use [their personal favorite tool]", laugh in their face and walk away from that conversation.__
+
+## Setting Up
+The relationship between VS Code and Ruby is still in its infancy, so many of the tools that we regularly use in other languages (such as C++, Javascript or Ook) aren't immediately accessible. Our process will be:
+
+1. Set up the file we want to debug
```
Comment (tildeee) [inline 00-programming-fundamentals/setting-up-vs-code-debugger.md]: This step (and the `my_efficient_and_clever_program.rb`) doesn't seem to be relevant to this setup exercise, is it possible to move this step? Basically, the code snippet below doesn't get used until the next activity, and it may be good to keep those bundled together.

Perhaps there are steps for setting up, and then steps outlined in the next activity includes "set up the file we want to debug."

File: 00-programming-fundamentals/setting-up-vs-code-debugger.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+1. Make sure we have the proper gems on our machine
+1. Configure our `launch.json` in our `.vscode` folder
+1. Set our breakpoint
+1. Debug our code
+
+Copy and paste the following code into a new `.rb` file:
```
Comment (tildeee) [inline 00-programming-fundamentals/setting-up-vs-code-debugger.md]: Maybe we can just give them a concrete Ruby file name at this point, too (`my_efficient_and_clever_program.rb` is what's named in the comment below)
